flecs_ecs/tests/entity_test.rs

```diff
@@ -1720,23 +1719,20 @@ fn entity_defer_set_3_components() {
 fn entity_set_2_w_on_set() {
     let world = create_world();
 
-    let mut position_set = 0;
-    let mut velocity_set = 0;
-
     world
-        .observer::<(&Position,)>()
+        .observer::<&Position>()
         .add_event::<flecs::OnSet>()
-        .each(|(p,)| {
-            position_set += 1;
+        .each_entity(|entity, p| {
+            entity.world().get_mut::<Flags>().insert("position_set", 1);
```

Comment:
This changes the test, can you make it so it counts it instead? I've actually had a bug before where it was set twice 😅